php/class-post-type.php

```diff
@@ -549,4 +579,147 @@ public function filter_user_has_cap( $allcaps, $caps ) {
+
+			/** This action is documented in wp-includes/class-wp-customize-manager.php */
+			do_action( 'customize_register', $this->snapshot_manager->customize_manager );
+		}
+		$snapshot_content = $this->get_post_content( $post );
+
+		if ( method_exists( $this->snapshot_manager->customize_manager, 'validate_setting_values' ) ) {
+			/** This action is documented in wp-includes/class-wp-customize-manager.php */
+			do_action( 'customize_save_validation_before', $this->snapshot_manager->customize_manager );
+		}
+
+		$this->snapshot_manager->customize_manager->add_dynamic_settings( array_keys( $snapshot_content ) );
+		$setting_objs = array();
+		$have_error = false;
+		foreach ( $snapshot_content as $setting_id => &$setting_params ) {
+			if ( ! isset( $setting_params['value'] ) || is_null( $setting_params['value'] ) ) {
+				// Null setting save error.
+				if ( ! is_array( $setting_params ) ) {
+					if ( ! empty( $setting_params ) ) {
+						$setting_params = array( 'value' => $setting_params );
+					} else {
+						$setting_params = array();
+					}
+				}
+				$setting_params['publish_error'] = 'null_value';
+				$have_error = true;
+			}
+			$this->snapshot_manager->customize_manager->set_post_value( $setting_id, $setting_params['value'] );
+			$setting_obj = $this->snapshot_manager->customize_manager->get_setting( $setting_id );
+			if ( $setting_obj instanceof \WP_Customize_Setting ) {
+				$setting_objs[] = $setting_obj;
+			} elseif ( ! isset( $setting_params['publish_error'] ) ) {
+				// Invalid setting save error.
+				$setting_params['publish_error'] = 'setting_object_not_found';
+				$have_error = true;
+			}
+		}
+		if ( empty( $setting_objs ) ) {
+			$update_setting_args = array(
+				'ID' => $post->ID,
+				'post_status' => 'pending',
+			);
+			if ( true === $have_error ) {
+				$update_setting_args['post_content'] = Customize_Snapshot_Manager::encode_json( $snapshot_content );
+			}
+			wp_update_post( $update_setting_args );
+			update_post_meta( $post_id, 'snapshot_error_on_publish', 1 );
+			return;
+		}
+
+		if ( true === $have_error ) {
+			$this->suppress_publish_hook = true;
+			wp_update_post( array(
+				'ID' => $post->ID,
+				'post_content' => Customize_Snapshot_Manager::encode_json( $snapshot_content ),
+			) );
+			$this->suppress_publish_hook = false;
+		} else {
+			// Remove any previous error on setting.
+			delete_post_meta( $post_id, 'snapshot_error_on_publish' );
+		}
+
+		$existing_caps = wp_list_pluck( $setting_objs, 'capability' );
+		foreach ( $setting_objs as $setting_obj ) {
+			$setting_obj->capability = 'exist';
+		}
+
+		$action = 'customize_save';
+		$callback = array( $this->snapshot_manager, 'check_customize_publish_authorization' );
+		$priority = has_action( $action, $callback );
+		if ( false !== $priority ) {
+			remove_action( $action, $callback, $priority );
+		}
+		$this->snapshot_manager->customize_manager->doing_ajax( 'customize_save' );
```

Comment:
The `doing_ajax` method doesn't do anything other than return a `bool` to say whether this Ajax action is happening. So I don't think this block is doing anything. I'm revising.